services/wordpress/wordpress-rating.service.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+const BaseWordpress = require('./wordpress-base')
+const { starRating, metric } = require('../../lib/text-formatters')
+const { floorCount } = require('../../lib/color-formatters')
+
+const extensionData = {
+  plugin: {
+    capt: 'Plugin',
+    exampleSlug: 'bbpress',
+  },
+  theme: {
+    capt: 'Theme',
+    exampleSlug: 'twentyseventeen',
+  },
+}
+
+class WordpressRatingBase extends BaseWordpress {
+  static render({ response }) {
+    const total = response.num_ratings
+    const rating = (response.rating / 100) * 5
+    return {
+      message: `${rating}/5 (${metric(total)})`,
+      color: floorCount(rating, 2, 3, 4),
+    }
+  }
+
+  static get category() {
+    return 'rating'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'rating' }
+  }
+}
+
+function RatingForExtensionType(extensionType) {
+  const { capt, exampleSlug } = extensionData[extensionType]
+
+  return class WordpressRating extends WordpressRatingBase {
+    static get url() {
+      return {
+        base: `wordpress/${extensionType}/rating`,
+        format: '(.+)',
+        capture: ['slug'],
+      }
+    }
+
+    static get extensionType() {
+      return extensionType
+    }
+
+    static get examples() {
+      return [
+        {
+          title: `Wordpress ${capt} Rating`,
+          exampleUrl: exampleSlug,
+          urlPattern: ':slug',
+          staticExample: this.render({
+            response: {
+              rating: 80,
+              num_ratings: 100,
+            },
+          }),
+          keywords: ['wordpress'],
+        },
+      ]
+    }
+  }
+}
+
+function StarsForExtensionType(extensionType) {
+  const { capt, exampleSlug } = extensionData[extensionType]
+
+  return class WordpressStars extends WordpressRatingBase {
+    static render({ response }) {
+      const rating = (response.rating / 100) * 5
+      return { message: starRating(rating), color: floorCount(rating, 2, 3, 4) }
+    }
+
+    static get url() {
+      return {
+        base: `wordpress/${extensionType}`,
+        format: '(?:stars|r)/(.+)',
+        capture: ['slug'],
+      }
+    }
+
+    static get extensionType() {
+      return extensionType
+    }
+
+    static get examples() {
+      return [
+        {
+          title: `Wordpress ${capt} Rating`,
+          exampleUrl: exampleSlug,
+          urlPattern: '(r|stars)/:slug',
```

Comment:
Nearly there.. this bit should be:

```js
exampleUrl: `stars/${exampleSlug}`,
urlPattern: 'stars/:slug',
```

so that when we render the example page the pattern and example render as a working URL/pattern. There's no clever substitution going on here. We just concatenate `base` and `exampleUrl`/`urlPattern` so `${base}/${exampleUrl}` should give us a working example and `${base}/${urlPattern}` should give us a valid pattern. At the moment we've got this:

---

![screenshot at 2018-10-09 19-55-47](https://user-images.githubusercontent.com/6025893/46692383-06717d00-cbff-11e8-86d9-cb5bd999472b.png)

---

but we should be generating:

`/wordpress/plugin/stars/:slug.svg`
`/wordpress/plugin/stars/bbpress.svg`

Reply:
> Nearly there.. this bit should be:
> 
> ```js
> exampleUrl: `stars/${exampleSlug}`,
> urlPattern: 'stars/:slug',
> ```
Oh, I missed that. I will fix it shortly.

> but we should be generating:
> 
> `/wordpress/plugin/stars/:slug.svg`
> `/wordpress/plugin/stars/bbpress.svg`
 
I see I will do the same thing as [here](https://github.com/badges/shields/pull/2152/commits/3f70f9b91774c7dccf59e519f503228f5daa79e8#diff-beabaf484d8e9bdca8ef81b839688560R133) then